### ISSUE TITLE
Error messages when using 2 RectangularROI.

### ISSUE THREAD

**Issue by HanHsuanWu** [hyperspy/hyperspy#3213]
#### Describe the bug
Error message shows up when using 2 RectangularROI. The 2 RectangularROI still work as intended tho.

#### To Reproduce
The error message shows up when:
-  Change RectangularROI size.
-  Move RectangularROI.
- Double clicks on RectangularROI.

```
# minimal example
%matplotlib qt

import numpy as np
import matplotlib.pyplot as plt 
import hyperspy.api as hs

s = hs.signals.Signal1D(np.random.random((20, 20, 30)))

xx2, yy2, xx1, yy1 = 0, 0, 2, 2

shiftx = 5
shifty = 3

rect_roi0 = hs.roi.RectangularROI(xx2, yy2, xx1, yy1) # x1, y1, x2, y2
rect_roi1 = hs.roi.RectangularROI(xx2 + shiftx , yy2 + shifty, xx1 + shiftx, yy1 + shifty)

s.plot()

color = ["tab:blue", "tab:orange"]
roi0 = rect_roi0.interactive(s, color=color[0])
roi1 = rect_roi1.interactive(s, color=color[1])
```

#### Expected behavior
Error message not showing.

**Comment by ericpre** [comment]
The error occurs when selecting/deselecting the widget and the traceback is 

```python
Traceback (most recent call last):
  File "~\mambaforge\lib\site-packages\matplotlib\cbook\__init__.py", line 309, in process
    func(*args, **kwargs)
  File "~\Dev\hyperspy\hyperspy\drawing\widget.py", line 724, in onpick
    super(ResizableDraggableWidgetBase, self).onpick(event)
  File "~\Dev\hyperspy\hyperspy\drawing\widget.py", line 479, in onpick
    super(DraggableWidgetBase, self).onpick(event)
  File "~\Dev\hyperspy\hyperspy\drawing\widget.py", line 1004, in onpick
    self._set_resizers(False, self.ax)
  File "~\Dev\hyperspy\hyperspy\drawing\widget.py", line 904, in _set_resizers
    r.remove()
  File "~\mambaforge\lib\site-packages\matplotlib\artist.py", line 242, in remove
    self._remove_method(self)
ValueError: list.remove(x): x not in list
```

**Comment by ericpre** [comment]
Fixed in #3222.

**Comment by HanHsuanWu** [comment]
Wow thank you Eric!Sent from my iPhoneOn Sep 2, 2023, at 3:56 AM, Eric Prestat ***@***.***> wrote:﻿
Closed #3213 as completed.
